backend/src/airgap/detect.rs

```diff
@@ -59,138 +53,92 @@ impl AirgapDetection {
-                        error!("Failed to receive air gap status change: {e:#?}");
-                        break;
-                    }
-                }
-            }
-        });
-
-        std::thread::spawn({
-            let airgap_detection = airgap_detection.clone();
-
-            move || {
-                loop {
-                    airgap_detection.perform_detection();
-
-                    std::thread::sleep(Duration::from_secs(AIRGAP_DETECTION_INTERVAL));
-                }
+                inner_airgap_detection.perform_detection().await;
+                tokio::time::sleep(Duration::from_secs(AIRGAP_DETECTION_INTERVAL)).await;
             }
         });
 
         airgap_detection
     }
 
-    fn send_update(&self) {
-        if let Some(sender) = self.updates.as_ref() {
-            let status = if self.violation_detected() {
-                AirGapStatusChange::AirGapViolationDetected
-            } else {
-                AirGapStatusChange::AirGapViolationResolved
-            };
+    async fn log_status_change(&self) {
+        let event = if self.violation_detected() {
+            AuditEvent::AirGapViolationDetected
+        } else {
+            AuditEvent::AirGapViolationResolved
+        };
 
-            if let Err(e) = sender.send(status) {
-                error!("Failed to send airgap status update: {e}");
+        if let Some(pool) = &self.pool {
+            if let Err(e) = crate::audit_log::create(pool, &event, None, None, None).await {
+                error!("Failed to log air gap status change: {e:#?}");
             }
         }
     }
 
-    fn perform_detection(&self) {
+    async fn perform_detection(&self) {
         let was_connected = self.violation_detected();
 
-        if self.is_connected() {
+        if self.is_connected().await {
             self.set_airgap_violation_detected(true);
 
             if was_connected {
                 warn!("Airgap violation detected, abacus is still connected to the internet.");
             } else {
                 error!("Airgap violation detected, abacus is connected to the internet!");
-                self.send_update();
+                self.log_status_change().await;
             }
         } else {
             self.set_airgap_violation_detected(false);
 
             if was_connected {
                 info!("Airgap violation resolved, abacus is no longer connected to the internet.");
-                self.send_update();
+                self.log_status_change().await;
             } else {
                 trace!("No airgap violation detected.");
             }
         }
-
-        self.set_last_check();
     }
 
     /// Detects if the system is in an airgap by attempting to connect to IPv4, IPv6 addresses and performing DNS lookups
-    fn is_connected(&self) -> bool {
-        // attempt to connect to known IPv4 addresses over TCP
-        let tcp_ipv4_connection_success = IPV4
+    async fn is_connected(&self) -> bool {
```

Comment:
Maybe it's easier to spawn all connection/lookup tasks in parallel, e.g. using a [JoinSet](https://docs.rs/tokio/latest/tokio/task/struct.JoinSet.html)?

Reply:
Sure - see [a88def8](https://github.com/kiesraad/abacus/pull/1859/commits/a88def8e2352f327c81fc5d7e0f7d85888220492)